languages/scala/recursive_descent/Parser_scala_recursive_descent.ml

```diff
@@ -456,6 +456,38 @@ let lookingAhead2 body1 body2 in_ =
       let res2 = body2 in_' in
       res1 && res2
 
+(* ------------------------------------------------------------------------- *)
+(* indentation *)
+(* ------------------------------------------------------------------------- *)
+
+(* We used to just check the INDENT and DEDENT tokens, but this causes issues in
+   cases like this:
+
+   class Foo:
+     def foo() =
+       3
+
+     def bar() = 3
+
+   We ideally want to keep scanning the contents of Foo until we find a DEDENT,
+   which presumably signals the end of the class block. Unfortunately, we might
+   run into a DEDENT which is not the end of the class, such as the DEDENT
+   published right before the second "def", due to the indentation difference
+   between it and the 3.
+
+   So instead, we can just check the columns. We follow a rule that the first token
```

Comment:
Is it an official Scala rule?

Reply:
It's actually more like how I do it now, where we check the columns _as they relate to previous indentation regions_.